frontend/internal-packages/agent/src/qa-agent/testcaseGeneration/prompts.ts

```diff
@@ -1,48 +1,349 @@
+  'Should successfully delete user'
+);
+SELECT * FROM finish();
+
+### Example 2: Foreign Key Constraint on DELETE (Failure Case)
+SELECT plan(2);
+SELECT lives_ok(
+  $$INSERT INTO orders (user_id, product_id, quantity) VALUES (1, 1, 5)$$,
+  'Setup: Insert order referencing user'
+);
+SELECT throws_ok(
+  $$DELETE FROM users WHERE id = 1$$,
+  '23503'
+);
+SELECT * FROM finish();
+
+### Example 3: Conditional DELETE
+SELECT plan(3);
+SELECT lives_ok(
+  $$INSERT INTO expired_sessions (user_id, created_at) VALUES (1, NOW() - INTERVAL '2 days')$$,
+  'Setup: Insert expired session'
+);
+SELECT lives_ok(
+  $$DELETE FROM expired_sessions WHERE created_at < NOW() - INTERVAL '1 day'$$,
+  'Should delete expired sessions'
+);
+SELECT is(
+  (SELECT COUNT(*) FROM expired_sessions WHERE user_id = 1),
+  0::bigint,
+  'Expired session should be deleted'
+);
+SELECT * FROM finish();
+
+### Example 4: CASCADE DELETE Verification
+SELECT plan(3);
+SELECT lives_ok(
+  $$INSERT INTO users (name, email) VALUES ('CascadeTest', 'cascade@example.com')$$,
+  'Setup: Insert user'
+);
+SELECT lives_ok(
+  $$INSERT INTO user_profiles (user_id, bio)
+     VALUES ((SELECT id FROM users WHERE email = 'cascade@example.com'), 'Test bio')$$,
+  'Setup: Insert profile'
+);
+SELECT lives_ok(
+  $$DELETE FROM users WHERE email = 'cascade@example.com'$$,
+  'Should cascade delete user and profile'
+);
+SELECT * FROM finish();
+`
+
+const SELECT_EXAMPLES = `
+## SELECT Test Examples
+
+### Example 1: Simple Data Verification
+SELECT plan(1);
+SELECT is(
+  (SELECT COUNT(*) FROM users WHERE email LIKE '%@example.com'),
+  5::bigint,
+  'Should find 5 users with example.com email'
+);
+SELECT * FROM finish();
+
+### Example 2: Join Query Validation
+SELECT plan(1);
+SELECT is(
+  (SELECT COUNT(*)
+   FROM orders o
+   INNER JOIN users u ON o.user_id = u.id
+   WHERE u.email = 'alice@example.com'),
+  3::bigint,
+  'Alice should have 3 orders'
+);
+SELECT * FROM finish();
+
+### Example 3: Aggregate Function Test
+SELECT plan(1);
+SELECT ok(
+  (SELECT AVG(price) FROM products WHERE category = 'Electronics') > 100,
+  'Average electronics price should be over 100'
+);
+SELECT * FROM finish();
+
+### Example 4: Result Set Comparison
+SELECT plan(1);
+SELECT results_eq(
+  $$SELECT name FROM users WHERE active = true ORDER BY name$$,
+  $$VALUES ('Alice'), ('Bob'), ('Charlie')$$,
+  'Should return active users in alphabetical order'
+);
+SELECT * FROM finish();
+
+### Example 5: Complex Business Logic Validation
+SELECT plan(2);
+SELECT ok(
+  (SELECT COUNT(*) FROM orders WHERE status = 'pending' AND created_at < NOW() - INTERVAL '7 days') = 0,
+  'No orders should be pending for more than 7 days'
+);
+SELECT ok(
+  (SELECT SUM(quantity * price) FROM order_items WHERE order_id = 1) =
+  (SELECT total_amount FROM orders WHERE id = 1),
+  'Order total should match sum of item prices'
+);
+SELECT * FROM finish();
+`
+
+const BEST_PRACTICES = `
+## Best Practices
```

Comment:
I considered suggesting things like writing in the AAA pattern or Vladimir Khorikov's four pillars of good unit testing, but it seems better to let you experiment with these going forward.